workspaces/templates-lib/packages/infra-aws/src/awsUserUtils.ts

```diff
@@ -0,0 +1,193 @@
+  const ecsCredentials = new AWS.ECSCredentials({
+    httpOptions: { timeout: 5000 },
+    maxRetries: 10, // retry 10 times
+  });
+  await ecsCredentials.getPromise();
+
+  AWS.config.credentials = ecsCredentials;
+
+  if (!process.env.AWS_REGION) {
+    throw new Error(
+      'AWS region environment variable ("AWS_REGION") not defined for ECS task.'
+    );
+  }
+
+  AWS.config.update({ region: process.env.AWS_REGION });
+
+  return ecsCredentials;
+}
+
+export async function getAWSUserFromDefaultLocalProfile(): Promise<AWS.Credentials> {
+  let credentials = new AWS.SharedIniFileCredentials();
+
+  // if no access key is found, try loading process_credentials
+  if (!credentials.accessKeyId) {
+    credentials = new AWS.ProcessCredentials();
+    await credentials.refreshPromise();
+  }
+
+  AWS.config.credentials = credentials;
+  // see https://github.com/aws/aws-sdk-js/pull/1391
+  process.env.AWS_SDK_LOAD_CONFIG = 'true';
+  return credentials;
+}
+
+export async function getAWSUserFromGoldstackConfig(
+  config: AWSConfiguration,
+  userName: string
+): Promise<AWS.Credentials> {
+  const user = config.users.find((user) => user.name === userName);
+  if (!user) {
+    throw new Error(`User '${userName}' does not exist in AWS configuration.`);
+  }
+
+  if (user.type === 'profile') {
+    const userConfig = user.config as AWSProfileConfig;
+
+    if (process.env.AWS_SHARED_CREDENTIALS_FILE) {
+      console.warn(
+        `Using AWS_SHARED_CREDENTIALS_FILE environment variable: '${process.env.AWS_SHARED_CREDENTIALS_FILE}'. awsCredentialsFileName in configuration will be ignored.`
+      );
+    }
+
+    const envVarValues = {
+      AWS_SDK_LOAD_CONFIG: process.env.AWS_SDK_LOAD_CONFIG,
+      AWS_SHARED_CREDENTIALS_FILE: process.env.AWS_SHARED_CREDENTIALS_FILE,
+      AWS_CONFIG_FILE: process.env.AWS_CONFIG_FILE,
+    };
+    if (userConfig.awsConfigFileName) {
+      // support loading from both `config` and `credentials` files, see https://github.com/goldstack/goldstack/issues/17#issuecomment-1044811805
+      process.env.AWS_SDK_LOAD_CONFIG = '1';
+      // filename property is ignored if AWS_SDK_LOAD_CONFIG is set; thus need to set AWS_SHARED_CREDENTIALS_FILE.
+      process.env.AWS_SHARED_CREDENTIALS_FILE =
+        userConfig.awsCredentialsFileName;
+      process.env.AWS_CONFIG_FILE = userConfig.awsConfigFileName;
+    }
+
+    let credentials: AWS.Credentials;
+    let filename: string | undefined;
+    if (userConfig.awsConfigFileName) {
+      filename = undefined;
+    } else if (!process.env.SHARE_CREDENTIALS_FILE) {
+      filename = userConfig.awsCredentialsFileName;
+    }
+
+    if (userConfig.credentialsSource !== 'process') {
+      credentials = new AWS.SharedIniFileCredentials({
+        profile: userConfig.profile,
+        filename: filename,
+      });
+    } else {
+      credentials = new AWS.ProcessCredentials({
+        profile: userConfig.profile,
+        filename: filename,
```

Comment:
We could technically do this change with the env vars above but it feels like a quirk tightly coupled to the AWS JS SDK V2 implementation of `AWS.ProcessCredentials`.

```suggestion
    } else {
        // Allow `AWS.ProcessCredentials` to search the default config location `~/.aws/config` in addition to `credentials` 
        // This matches most other CLI / SDK implementations (including AWS JS SDK v3) and the behaviour of most `credential_process` helper tools
        // With this enabled, `AWS_CONFIG_FILE` must not contains an invalid path, but `AWS_SHARED_CREDENTIALS_FILE` can be missing.
        process.env.AWS_SDK_LOAD_CONFIG = '1';
        
        credentials = new AWS.ProcessCredentials({
        profile: userConfig.profile,
        filename: filename,
```

Reply:
NB. Given this refers to default locations, I'm not sure how to create an isolated unit test that covers it. 

I tested the following two commands to verify locally with my `~/.aws/config` setup but it's difficult to tell if there might be side-effects.

`yarn infra init dev` 🚫
`AWS_SDK_LOAD_CONFIG=1 yarn infra init dev` ✅

... using the following config:

```json
...
    {
      "name": "awsUser",
      "type": "profile",
      "config": {
        "profile": "account.Role",
        "awsDefaultRegion": "eu-west-1",
        "credentialsSource": "process"
      }
    }
...
```